### ISSUE TITLE
expected: map([...] (...) -> expected<...> { ... }) produces expected<expected>

### ISSUE THREAD

**Issue by ArsenArsen** [managarm/frigg#41]
While this behavior is probably fine, it would be desirable to be able to chain expecteds for monadic operations.

For an example of such see [Haskell's `Either`](https://hackage.haskell.org/package/base-4.17.0.0/docs/Data-Either.html)

```hs
-- Module      :  Data.Either
-- Copyright   :  (c) The University of Glasgow 2001
-- License     :  BSD-style (see the file libraries/base/LICENSE)

-- | @since 4.4.0.0
instance Monad (Either e) where
    Left  l >>= _ = Left l
    Right r >>= k = k r
```

In essence, this means that `>>=` propagates error and does not evaluate `k` unless the `Either` is a `Right` (success case)

**Comment by avdgrinten** [comment]
This functionality is useful, of course, but it'd rather have it in a separate function (other than `map()`) such that the function that is passed in can *always* be assumed to return a `frg::expected`.

**Comment by ArsenArsen** [comment]
Agreed, that's what I meant by "While this behavior is probably fine, it would be desirable to be able to chain expecteds for monadic operations". A sensible name might be `then` or `chain`
